src/app/embed/adapters/adapter.properties.ts

```diff
@@ -0,0 +1,47 @@
+import { Observable } from 'rxjs/Observable';
+
+
+export const PropNames = [
+  'audioUrl',
+  'title',
+  'subtitle',
+  'subscribeUrl',
+  'subscribeTarget',
+  'artworkUrl',
+  'feedArtworkUrl'
+]
+
+export interface AdapterProperties { 
+  audioUrl?: string;
+  title?: string;
+  subtitle?: string;
+  subscribeUrl?: string;
+  subscribeTarget?: string;
+  artworkUrl?: string;
+  feedArtworkUrl?: string;
+} 
+
+export interface DataAdapter {
+  getProperties: (params:Object) => Observable<AdapterProperties>
+}
+
+export function hasMinimumParams(props): boolean {
+  return (props.audioUrl !== undefined) &&
+    (props.title !== undefined) &&
+    (props.subtitle !== undefined) &&
+    (props.subscribeUrl !== undefined) &&
+    (props.subscribeTarget !== undefined) &&
+    (props.artworkUrl !== undefined)
+}
+export function getMergedValues(...data: AdapterProperties[]):AdapterProperties {
```

Comment:
It's confusing that this needs to be passed in reverse order (I know how we got here, but let's reverse here)